packages/core/src/Primitive.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import type { CommonProps } from './types';
+
+export type PrimitiveOwnProps<
+  E extends React.ElementType = React.ElementType
+> = {
+  as?: E;
+} & CommonProps;
+
+export type PrimitiveProps<E extends React.ElementType> = PrimitiveOwnProps<E> &
+  Omit<React.ComponentProps<E>, keyof PrimitiveOwnProps>;
+
+export type PolymorphicComponentProps<E extends React.ElementType, P> = P &
+  PrimitiveProps<E>;
+
+export type PolymorphicComponent<P, D extends React.ElementType = 'div'> = <
```

Comment:
Curious about `= 'div'` part, is that the default element type?

Reply:
yes, it's a default tag